GoogleForCreators/web-stories-wp
# Implement speculative prerendering

## Summary

This PR makes use of the [Speculation Rules API](https://developer.mozilla.org/en-US/docs/Web/API/Speculation_Rules_API) to enable dynamically prerendering editor and story URLs.

## User-facing changes


Prerendering for these URLs has been enabled for the following pages:
1. Dashboard - Create New Story, Edit story
2. All Stories - Create New Story, Edit story, View story

## Testing Instructions

This PR can be tested by following these steps:

1. Go to the Stories Dashboard page.
2. Open the Application Tab and go to Speculative loads > Speculations.
3. A list of URLs with their Status should appear.
4. Hover over a story to see the status change to Ready.

## WIP

1. Add prerendering on the front end for story permalink when hovering over the link to view the story
2. Add a check to see if the Speculation Rules plugin is active


## Reviews

### Does this PR have a security-related impact?
No.

### Does this PR change what data or activity we track or use?
🤔

### Does this PR have a legal-related impact?
No.

## Checklist


---

Fixes #13603
